x-pack/plugins/fleet/server/saved_objects/index.ts

```diff
@@ -269,6 +272,16 @@ const getSavedObjectTypes = (): { [key: string]: SavedObjectsType } => ({
         created_by: { type: 'keyword' },
       },
     },
+    modelVersions: {
+      '1': {
+        changes: [
+          {
+            type: 'data_backfill',
+            backfillFn: migratePackagePolicyToV8100,
+          },
+        ],
+      },
```

Comment:
```
policy.windows.behavior_protection.reputation_service = false;
policy.mac.behavior_protection.reputation_service = false;
policy.linux.behavior_protection.reputation_service = false;
```

How will the previous version of Kibana react if it encounters documents with those new properties? Will it be able to function properly or not?

If not, you will need to implement a `forwardCompatibility` schema

https://github.com/elastic/kibana/blob/fd068da3a4e7759cc636831d01e99f17958690be/packages/core/saved-objects/core-saved-objects-server/src/model_version/schemas.ts#L18-L32

Reply:
@pgayvallet for the `forwardCompatibility` - how does this work?  Does it simply modify the data that is returned to the earlier Kibana?  It doesn't modify the actual underlying doc/SO correct?  It's important we don't actually change those since they represent Agent Policies that should retain their values unless changed by users.

If this is the case, then we would just need to write a `forwardCompatibility` schema which would validate that those fields are not present.

Reply:
> Does it simply modify the data that is returned to the earlier Kibana? It doesn't modify the actual underlying doc/SO correct?

Correct. the purpose is to shallow newly added fields to older versions when necessary (when the newly introduced fields could causes errors or problems for the previous version)

> If this is the case, then we would just need to write a forwardCompatibility schema which would validate that those fields are not present.

It shouldn't validate their presence, it should evict them. I encourage you to read [the model version guide](https://github.com/elastic/kibana/blob/52eb488692ca74cdc388188100540cc39e5d887e/packages/core/saved-objects/core-saved-objects-server/docs/model_versions.md#schemas) for more info.